lib/bootloader_pvm.pm

```diff
@@ -112,13 +100,43 @@ sub prepare_pvm_installation {
     type_string_slow " fips=1" if (get_var('FIPS_INSTALLATION'));
     type_string_slow " UPGRADE=1" if (get_var('UPGRADE'));
     send_key 'ret';
-
     assert_screen "pvm-grub-command-line-fresh-prompt", 180, no_wait => 1;    # kernel is downloaded while waiting
     enter_cmd_slow "initrd $mntpoint/initrd";
+}
 
-    assert_screen "pvm-grub-command-line-fresh-prompt", 180, no_wait => 1;    # initrd is downloaded while waiting
-    enter_cmd "boot";
-    save_screenshot;
+sub prepare_pvm_installation {
+    my ($boot_attempt) = @_;
+    $boot_attempt //= 1;
+    # the grub on powerVM has a rather strange feature that it will boot
+    # into the firmware if the lpar was reconfigured in between and the
+    # first menu entry was used to enter the command line. So we need to
+    # reset the LPAR manually
+    if (match_has_tag('novalink-failed-first-boot')) {
+        enter_cmd "set-default ibm,fw-nbr-reboots";
+        enter_cmd "reset-all";
+        assert_screen 'pvm-firmware-prompt';
+        send_key '1';
+        get_into_net_boot;
+    }
+    enter_netboot_parameters;
+    # we have sporadic netwwork issue on power machine 'grenache', mnt directory cannot be accessed, see poo#120570
+    # record_soft_failure if error occurs while loading initrd
+    assert_screen([qw(pvm-grub-command-line-fresh-prompt load-initrd-error)], 180, no_wait => 1);    # initrd is downloaded while waiting
+    if (match_has_tag 'load-initrd-error') {
+        record_info 'cannot load initrd, sporadic issue, see poo#120570. Re-trying now...';
+        enter_cmd_slow "exit";
+        enter_cmd_slow "reset-all";
+        send_key '1';
```

Comment:
Maybe you also want to `assert_screen "pvm-firmware-prompt"` and even better, since the steps seem to be the same, you could make a new function, and use it in both places, since the code is (almost) the same

Reply:
yes, I will change this now, thanks!